pkg/gen/awaitComments.go

```diff
@@ -15,15 +15,46 @@
 package gen
 
 import (
+	"fmt"
+
 	"github.com/pulumi/pulumi-kubernetes/pkg/kinds"
 )
 
-const preamble = `This resource waits until it is ready before registering success for
+func timeoutComment(kind kinds.Kind) string {
+	const timeoutOverride = `setting 'pulumi.com/timeoutSeconds' as a '.metadata.annotation' on the resource.`
+
+	timeout := func(kind kinds.Kind) string {
```

Comment:
Do we have any referable reasons as to why we're using the following defaults?

I lean towards setting all timeouts to a minimum default of 10 minutes, particularly for Pods, Deployments and StatefulSets since many factors can delay Pod readiness:
- Slow connection
- Slow registry
- Large Docker images being pulled
- Readiness / Liveness Probes
- PodDisruptionBudgets that take a step-wise approach to fan-outs & fan-ins

Reply:
I don't think there was any empirical reason. I'm fine bumping them to 10 mins.

@hausdorff Thoughts?

Reply:
I decided to pull the behavioral change out into a separate PR, and only update docs in this PR.